### PR TITLE
 [FLINK-17004] Document the LIKE clause of CREATE TABLE statement. 

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

Document the LIKE clause of CREATE TABLE statement.  It is based on #11981

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

**Comment by dawidwys** [comment]
cc @aljoscha

File: docs/dev/table/sql/create.md [hunk 1]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
 
 **Notes:** The table registered with `CREATE TABLE` statement can be used as both table source and table sink, we can not decide if it is used as a source or sink until it is referenced in the DMLs.
 
+**LIKE clause**
+
+The `LIKE` clause is a variant of a SQL features (Feature T171, “LIKE clause in table definition” and Feature T173, “Extended LIKE clause in table definition”). The clause can be used to create a table based on a definition of an existing table. Additionally, users
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:220]:
```suggestion
The `LIKE` clause is a variant of a SQL feature (Feature T171, “LIKE clause in table definition” and Feature T173, “Extended LIKE clause in table definition”). The clause can be used to create a table based on a definition of an existing table. Additionally, users
```

File: docs/dev/table/sql/create.md [hunk 2]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
...
+can extend the original table or exclude certain parts of it. In contrast to the SQL standard the clause must be defined at the top-level of a CREATE statement. That is because the clause applies to multiple parts of the definition and not only to the schema part.
+
+You can use the clause e.g. to reuse, but overwrite certain connector properties or add watermarks to tables defined externally, e.g. add a watermark to a table created in Apache Hive.
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:223]:
```suggestion
You can use the clause e.g. to reuse (and potentially overwrite) certain connector properties or add watermarks to tables defined externally, e.g. add a watermark to a table created in Apache Hive. 
```

File: docs/dev/table/sql/create.md [hunk 3]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
...
+* EXCLUDING - Does not include the given feature of the source table.
+* OVERWRITING - Includes the feature of the source table, overwrites duplicate entries of the source table with properties of the new table, e.g. if an option with the same key exists in both tables, the one from the current statement will be used.
+
+Additionally, you can use the `INCLUDING/EXCLUDING ALL` option to specify what should be the strategy if there was no specific strategy defined, i.e. if you use `EXCLUDING ALL INCLUDING WATERMARKS` only the watermarks will be included from the source table.
+
+If you provide no like options, by default the planner will use `INCLUDING ALL OVERWRITING OPTIONS` options.
+
+**NOTE** You cannot control the behavior of merging physical fields. Those will be merged as if you applied the `INCLUDING` strategy.
+
+{% highlight sql %}
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:281]: This example seems to be dangling at the end.

File: docs/dev/table/sql/create.md [hunk 4]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
 
 **Notes:** The table registered with `CREATE TABLE` statement can be used as both table source and table sink, we can not decide if it is used as a source or sink until it is referenced in the DMLs.
 
+**LIKE clause**
+
+The `LIKE` clause is a variant of SQL features (Feature T171, “LIKE clause in table definition” and Feature T173, “Extended LIKE clause in table definition”). The clause can be used to create a table based on a definition of an existing table. Additionally, users
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:220]:
```suggestion
The `LIKE` clause is a variant/combination of SQL features (Feature T171, “LIKE clause in table definition” and Feature T173, “Extended LIKE clause in table definition”). The clause can be used to create a table based on a definition of an existing table. Additionally, users
```

File: docs/dev/table/sql/create.md [hunk 5]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
...
+can extend the original table or exclude certain parts of it. In contrast to the SQL standard the clause must be defined at the top-level of a CREATE statement. That is because the clause applies to multiple parts of the definition and not only to the schema part.
+
+You can use the clause e.g. to reuse (and potentially overwrite) certain connector properties or add watermarks to tables defined externally, e.g. add a watermark to a table created in Apache Hive.
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:223]:
```suggestion
You can use the clause to reuse (and potentially overwrite) certain connector properties or add watermarks to tables defined externally. For example, you can add a watermark to a table defined in Apache Hive. 
```

I think we're going a bit overboard with `e.g.` 😅

File: docs/dev/table/sql/create.md [hunk 6]
```diff
@@ -208,6 +215,101 @@ The key and value of expression `key1=val1` should both be string literal. See d
...
+)
+LIKE Orders_in_file (
+    -- Exclude everything besides the computed columns which we need to generate the watermark for.
+    -- We do not want to have the partitions or filesystem options as those do not apply to kafka. 
+    EXCLUDING ALL
+    INCLUDING GENERATED
+);
+{% endhighlight %}
+
+If you provide no like options, by default the planner will use `INCLUDING ALL OVERWRITING OPTIONS` options.
```
Comment (aljoscha) [inline docs/dev/table/sql/create.md:309]:
```suggestion
If you provide no like options, `INCLUDING ALL OVERWRITING OPTIONS` will be used as a default.
```

I think it's an unnecessary here that the planner is involved.

Comment (dawidwys) [reply docs/dev/table/sql/create.md:309]: :+1: That was my try on using more active voice. I agree it's unnecessary here.

**Comment by yangyichao-mango** [comment]
Thx.
Maybe there is no need to copy to Chinese Version. I'm already translating.

**Pull request closed without merging**